signac/common/config.py

```diff
@@ -99,45 +98,33 @@ def read_config_file(filename, configspec=None, *args, **kwargs):
     return config
 
 
-def load_config(root=None, local=False):
-    """Load configuration, searching upward from a root path if desired.
+def load_config(root=None):
+    """Load configuration at root directory.
 
     Parameters
     ----------
     root : str
-        The path from which to begin searching for config files.
-    local : bool, optional
-        If ``True``, only search in the provided directory and do not traverse
-        upwards through the filesystem (Default value: False).
+        The path from which to load the local config.
 
     Returns
     --------
     :class:`Config`
         The composite configuration including both local and global config data
-        if requested.
+        if requested. Note that because this config is a composite,
+        modifications to the returned value will not be reflected in the files.
```

Comment:
Modifications to a `Config` object are _never_ reflected in a file, right? It's not like a synced collection iirc, but has methods to write to a file? Can't remember.

Reply:
The `configobj` API is pretty confusing in this respect. You are correct in the sense that it will not automatically reflect changes to a file. However, there `ConfigObj.write` behaves differently depending on whether you initialize the object with a filename or not. If you do, then `write` will create if necessary and write to the file. Otherwise, if you just do something like `c = ConfigObj(); c['a'] = 'b'; c.write()` it will just print the output. That means that a `Config` returned from `load_config` can't be written back to file, while a `Config` returned from `read_config_file` can be. I've been bitten by this many times in the past, so I wanted to add a note. I am open to a more robust solution in the long term that avoids this though, and I'm hopeful that whatever package we use for TOML support when we switch will allow us to avoid this problem. Alternatively, maybe in 2.0 we can remove all functions that return writeable configs and force any mutating code path to invoke the underlying TOML library directly. I'm not sure what the best approach is here, but for now I figured documentation was a start.

Reply:
Sounds good to me.